dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -58,64 +58,16 @@ static const columninfo RecycleBinColumns[] =
 
 #define COLUMNS_COUNT  6
 
-/*
- * Recycle Bin folder
- */
-
-HRESULT CRecyclerExtractIcon_CreateInstance(LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
+static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl)
 {
-    CComPtr<IDefaultExtractIconInit> initIcon;
-    HRESULT hr = SHCreateDefaultExtractIcon(IID_PPV_ARG(IDefaultExtractIconInit, &initIcon));
-    if (FAILED_UNEXPECTEDLY(hr))
-        return hr;
+    LPPIDLDATA pdata = _ILGetDataPointer(pidl);
 
-    /* FIXME: This is completely unimplemented */
-    initIcon->SetNormalIcon(swShell32Name, 0);
+    if (pdata && pdata->type == 0x00)
```

Comment:
Did you ever get a zero type during testing? 👀
(According to https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/wine/pidl.h;hb=f18fb7da0918f48310c9ceeaffe7df7d2e300017#l84
this would be `PT_CPLAPPLET` 🤔 )

Reply:
EDIT: I've just seen that it gets set like that by some other existing code in this file.

Reply:
yeah, that's why _ILGetRecycleStruct is used

Reply:
i can add a check for any case

Reply:
Done